models/spike_train_injector.h

```diff
@@ -0,0 +1,374 @@
+/*
+ *  spike_train_injector.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SPIKE_TRAIN_INJECTOR_H
+#define SPIKE_TRAIN_INJECTOR_H
+
+// C++ includes:
+#include <vector>
+
+// Includes from nestkernel:
+#include "connection.h"
+#include "device.h"
+#include "event.h"
+#include "nest_time.h"
+#include "nest_types.h"
+#include "node.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: neuron, device, spike, generator
+
+Short description
++++++++++++++++++
+
+Neuron that emits prescribed spike trains.
+
+Description
++++++++++++
+
+The spike train injector neuron emits spikes at prescribed spike times
+which are given as an array. The neuron does not allow incoming connections
+and is thus not able to process incoming spikes or currents.
+
+.. note::
+
+   ``spike_train_injector`` is recommended if the spike trains have a similar
+   rate to regular neurons. For very high rates, use
+   :doc:`spike generator </models/spike_generator>`.
+
+Spike times are given in milliseconds as an array. The `spike_times` array
+must be sorted with the earliest spike first. All spike times must be strictly
+in the future. Trying to set a spike time in the past or at the current time
+step will cause a NEST error. Setting a spike time of 0.0 will also result
+in an error.
+
+Multiple occurrences of the same time indicate that more than one event is to
+be generated at this particular time.
+
+The spike generator supports spike times that do not coincide with a time
+step, i.e., are not falling on the grid defined by the simulation resolution.
+There are three options that control how spike times that do not coincide
```

Comment:
Reword like 71/72

Spike times that do not coincide with a step are handled with one of three options (see examples below).